builtin/config.c

```diff
@@ -189,11 +189,41 @@ static void show_config_origin(struct strbuf *buf)
 	strbuf_addch(buf, term);
 }
 
+static void show_config_origin_type(struct strbuf *buf)
+{
+	const char *scope;
+	const char term = end_null ? '\0' : '\t';
+	switch (current_config_scope()) {
+	case CONFIG_SCOPE_SYSTEM:
+		scope = "(system)";
+		break;
+	case CONFIG_SCOPE_GLOBAL:
+		scope = "(global)";
+		break;
+	case CONFIG_SCOPE_REPO:
+		scope = "(local)";
+		break;
+	case CONFIG_SCOPE_CMDLINE:
+		scope = "(command line)";
+		break;
+	default:
+		scope = "(unknown)";
+		break;
+	}
+
+	if (end_null)
+		strbuf_addstr(buf, scope);
+	else
+		quote_c_style(scope, buf, NULL, 0);
+	strbuf_addch(buf, term);
```

Comment:
A slightly cleaner approach, I think, would be to change `show_config_origin_type()` to return the string instead, probably renaming it to `origin_type_to_string()`, so that it has only one concern, and is independent of the concern to write the output as part of `show_all_config()`.

Reply:
Normally I'd agree with you, I was just trying to match the style of `show_config_origin()`, but besides that there's no reason to not just return the string.

Reply:
Fair enough. I'd still contend that this is not a style issue, but that `show_config_origin()` is an example I'd rather not follow.